uint8 upper_red
uint8 upper_green
uint8 upper_blue
uint8 lower_red
uint8 lower_green
uint8 lower_blue

